### PR TITLE
Update the analytical solution for verification case ver-1e

### PR DESCRIPTION
**Pull request by lin-yang-ly**
Update the analytical solution of SiC and PyC layers for ver-1e in documentation and python script.

(Ref. #123)

### CODE REVIEW

File: doc/content/verification_and_validation/ver-1e.md [hunk 1]
```diff
@@ -63,13 +78,19 @@ and $\lambda_n$ are the roots of
...
+       style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
+       id=ver-1e_comparison_time_PyC
+       caption=Comparison of TMAP8 calculation with the analytical solution in PyC layer. Bold text next to the plot curves shows the RMSPE for the match between the TMAP8 prediction and analytical solution for the TMAP4 and TMAP7 verification cases.
 
 !media comparison_ver-1e.py
        image_name=ver-1e_comparison_time.png
        style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
        id=ver-1e_comparison_time
-       caption=Comparison of TMAP8 calculation with the analytical solution. Bold text next to the plot curves shows the RMSPE for the match between the TMAP8 prediction and analytical solution for the TMAP4 and TMAP7 verification cases.
+       caption=Comparison of TMAP8 calculation with the analytical solution in SiC layer. Bold text next to the plot curves shows the RMSPE for the match between the TMAP8 prediction and analytical solution for the TMAP4 and TMAP7 verification cases.
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1e.md:93]: Please specify the position of the point in the caption.

File: doc/content/verification_and_validation/ver-1e.md [hunk 2]
```diff
@@ -63,13 +78,19 @@ and $\lambda_n$ are the roots of
...
+
+!media comparison_ver-1e.py
+       image_name=ver-1e_comparison_time_PyC.png
+       style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
+       id=ver-1e_comparison_time_PyC
+       caption=Comparison of TMAP8 calculation with the analytical solution in PyC layer. Bold text next to the plot curves shows the RMSPE for the match between the TMAP8 prediction and analytical solution for the TMAP4 and TMAP7 verification cases.
```
[the remark this answers is not in the document]
Comment (lin-yang-ly) [reply doc/content/verification_and_validation/ver-1e.md:95]: Hello PC, I tried the deeper distance (x=32 nm, the interface is 33 nm) in PyC layer. However, the results reached the equilibrium immediately as well. The main reason is we have a very large diffusivity in PyC (D=1.274e-7), which is 10e4 times than diffusivity in SiC (2.622e-11). Also, the value is not accurate if I closeup to the time from 0 to 0.01 s for the comparison in PyC layer.

File: doc/content/verification_and_validation/ver-1e.md [hunk 3]
```diff
@@ -66,33 +70,37 @@ and $\lambda_n$ are the roots of
...
 
-The error is calculated between the TMAP8 and analytical solution values after $t$ = 0.2 s. This is in order to ignore the unphysical predictions of the analytical solution at very small times as shown in [ver-1e_comparison_time_zoomed], which is a close-up view of [ver-1e_comparison_time] close to the start of the simulation.
+The error is calculated between the TMAP8 and analytical solution values after $t$ = 0.01 s. This is in order to ignore the unphysical predictions of the analytical solution at very small times as shown in [ver-1e_comparison_time_zoomed], which is a close-up view of [ver-1e_comparison_time] close to the start of the simulation. The departure from physical results at low $t$ is due to the limited number of $\lambda_n$ values being used from [eqn:roots]. We use the $\lambda_n$ values from 0 to 100 in this case. The $\lambda_n$ values in larger range will only have limited improvement in the accuracy but increase the running time of the analytical solutions.
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1e.md:103]:
```suggestion
The error is calculated between the TMAP8 and analytical solution values after $t$ = 0.01 s. This is in order to ignore the unphysical predictions of the analytical solution at very small times as shown in [ver-1e_comparison_time_zoomed], which is a close-up view of [ver-1e_comparison_time] close to the start of the simulation. The departure from physical results at low $t$ is due to the limited number of $\lambda_n$ values being used from [eqn:roots]. We use the $\lambda_n$ values from 0 to 100 in this case. The $\lambda_n$ values in larger range will only have limited improvement in the accuracy but increase the running time for the computation of the analytical solutions.
```

File: doc/content/verification_and_validation/ver-1e.md [hunk 4]
```diff
@@ -22,56 +26,81 @@ where
 
     $x$ = distance from free surface of PyC
 
-    $a$ = thickness of the PyC layer (33 $\mu m$)
+    $a$ = thickness of the PyC layer (33 $\mu$ m)
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1e.md:29]:
```suggestion
    $a$ = thickness of the PyC layer (33 $\mu$m)
```

File: doc/content/verification_and_validation/ver-1e.md [hunk 5]
```diff
@@ -22,56 +26,81 @@ where
...
 
-    $l$ = thickness of the SiC layer (63 $\mu m$ in TMAP4 verification case, and 66 $\mu m$ in TMAP7 verification case)
+    $l$ = thickness of the SiC layer (63 $\mu$ m in TMAP4 verification case, and 66 $\mu$ m in TMAP7 verification case)
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1e.md:31]:
```suggestion
    $l$ = thickness of the SiC layer (63 $\mu$m in TMAP4 verification case, and 66 $\mu$m in TMAP7 verification case)
```

File: doc/content/verification_and_validation/ver-1e.md [hunk 6]
```diff
@@ -22,56 +26,81 @@ where
...
+The expressions of the analytical solution for the transient case in TMAP4 ([!cite](longhurst1992verification)) and TMAP7 ([!cite](ambrosek2008verification)) are inconsistent with the results, which suggest typographical errors. Moreover, no reference is provided. For these reasons, we use a different analytical transient solution and roots of $\lambda$ for the concentration in PyC and SiC layer from [!cite](li2010analytical).
 
 ## Results
 
-[ver-1e_comparison_dist] shows the comparison of the TMAP8 calculation and the analytical solution for concentration after steady-state is reached. The plots show the TMAP8 and analytical solution comparisons for both the TMAP4 case ($l$ = 63 $\mu m$) and the TMAP7 case ($l$ = 66 $\mu m$).
+[ver-1e_comparison_dist] shows the comparison of the TMAP8 calculation and the analytical solution for concentration after steady-state is reached. The plots show the TMAP8 and analytical solution comparisons for both the TMAP4 case ($l$ = 63 $\mu$ m) and the TMAP7 case ($l$ = 66 $\mu$ m).
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1e.md:77]:
```suggestion
[ver-1e_comparison_dist] shows the comparison of the TMAP8 calculation and the analytical solution for concentration after steady-state is reached. The plots show the TMAP8 and analytical solution comparisons for both the TMAP4 case ($l$ = 63 $\mu$m) and the TMAP7 case ($l$ = 66 $\mu$m).
```

File: doc/content/verification_and_validation/ver-1e.md [hunk 7]
```diff
@@ -22,56 +26,81 @@ where
...
+       caption=Comparison of TMAP8 calculation with the analytical solution at steady state. Bold text next to the plot curves shows the root mean square percentage error (RMSPE) between the TMAP8 prediction and analytical solution for the TMAP4 and TMAP7 verification cases.
 
-For transient solution comparison, the concentation at a point, which is $x$ $\mu m$ away from the PyC-SiC interface into the SiC layer, is obtained using the TMAP code as well as analytically. [ver-1e_comparison_time] shows comparison of the TMAP calculation with the analytical solution for this transient case. In the TMAP4 case, $x$ = 8 $\mu m$, and in the TMAP7 case $x$ = 15.75 $\mu m$. There is good agreement between TMAP and the analytical solution for both steady state as well as transient cases. In both cases, the root mean square percentage error (RMSPE) is under 0.2 %.
+For transient solution comparisons, we select two different points: One in the PyC layer at $x = 32$ $\mu$ m, and one in the SiC layer.
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1e.md:85]:
```suggestion
For transient solution comparisons, we select two different points: One in the PyC layer at $x = 32$ $\mu$m, and one in the SiC layer.
```

File: doc/content/verification_and_validation/ver-1e.md [hunk 8]
```diff
@@ -22,56 +26,81 @@ where
...
+The concentration at these two points is calculated using TMAP8 and with the analytical equations ([eqn:transient_PyC,eqn:transient_SiC]).
+[ver-1e_comparison_time_PyC] shows the comparison of the TMAP8 calculation against the analytical solution for this transient case in the PyC layer from 0 s to 1 s. Note that we only use the parameters from TMAP7 because the trivial difference between TMAP4 and TMAP7 in PyC layer.
+[ver-1e_comparison_time] shows the comparison of the TMAP8 calculation against the analytical solution in the SiC layer. In the TMAP4 case, $x$ = 41 $\mu$ m, and in the TMAP7 case, $x$ = 48.75 $\mu$ m.
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1e.md:88]:
```suggestion
[ver-1e_comparison_time] shows the comparison of the TMAP8 calculation against the analytical solution in the SiC layer. In the TMAP4 case, $x$ = 41 $\mu$m, and in the TMAP7 case, $x$ = 48.75 $\mu$m.
```

File: doc/content/verification_and_validation/ver-1e.md [hunk 9]
```diff
@@ -22,56 +26,81 @@ where
...
+There is good agreement between TMAP8 and the analytical solution for both steady state as well as transient cases. In both cases, the root mean square percentage error (RMSPE) is under 0.2 %.
+
+!media comparison_ver-1e.py
+       image_name=ver-1e_comparison_time_PyC.png
+       style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
+       id=ver-1e_comparison_time_PyC
+       caption=Comparison of TMAP8 calculation with the analytical solution at $x = 32$ $\mu$ m in PyC layer. Bold text next to the plot curves shows the RMSPE for the match between the TMAP8 prediction and analytical solution for the TMAP4 and TMAP7 verification cases.
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1e.md:95]:
```suggestion
       caption=Comparison of TMAP8 calculation with the analytical solution at $x = 32$ $\mu$m in PyC layer. Bold text next to the plot curves shows the RMSPE for the match between the TMAP8 prediction and analytical solution for the TMAP4 and TMAP7 verification cases.
```

File: doc/content/verification_and_validation/ver-1e.md [hunk 10]
```diff
@@ -22,56 +26,81 @@ where
...
 
 !media comparison_ver-1e.py
        image_name=ver-1e_comparison_time.png
        style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
        id=ver-1e_comparison_time
-       caption=Comparison of TMAP8 calculation with the analytical solution. Bold text next to the plot curves shows the RMSPE for the match between the TMAP8 prediction and analytical solution for the TMAP4 and TMAP7 verification cases.
+       caption=Comparison of TMAP8 calculation with the analytical solution in SiC layer. The compared distances are $x$ = 41 $\mu$ m in the TMAP4 case and $x$ = 48.75 $\mu$ m in the TMAP7 case. Bold text next to the plot curves shows the RMSPE for the match between the TMAP8 prediction and analytical solution for the TMAP4 and TMAP7 verification cases.
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1e.md:101]:
```suggestion
       caption=Comparison of TMAP8 calculation with the analytical solution in SiC layer. The compared distances are $x$ = 41 $\mu$m in the TMAP4 case and $x$ = 48.75 $\mu$m in the TMAP7 case. Bold text next to the plot curves shows the RMSPE for the match between the TMAP8 prediction and analytical solution for the TMAP4 and TMAP7 verification cases.
```

**Review by simopier — CHANGES_REQUESTED** [review]
Just minor changes. I'll need to doubele check the documentation, but otherwise things look great!
Thank you!

**Review by simopier — COMMENTED** [review]
The conflicts need to be addressed, the rest looks good!

File: doc/content/verification_and_validation/ver-1e.md [hunk 11]
```diff
@@ -30,48 +34,73 @@ where
...
 
-The error is calculated between the TMAP8 and analytical solution values after $t$ = 0.2 s. This is in order to ignore the unphysical predictions of the analytical solution at very small times as shown in [ver-1e_comparison_time_zoomed], which is a close-up view of [ver-1e_comparison_time] close to the start of the simulation.
+The error is calculated between the TMAP8 and analytical solution values after $t$ = 0.01 s. This is in order to ignore the unphysical predictions of the analytical solution at very small times as shown in [ver-1e_comparison_time_zoomed], which is a close-up view of [ver-1e_comparison_time] close to the start of the simulation. The departure from physical results at low $t$ is due to the limited number of $\lambda_n$ values being used from [eqn:roots]. We use the $\lambda_n$ values from 0 to 100 in this case. The $\lambda_n$ values in larger range will only have limited improvement in the accuracy but increase the running time for the computation of the analytical solutions.
 
 !media comparison_ver-1e.py
```
[the remark this answers is not in the document]
Comment (lin-yang-ly) [reply doc/content/verification_and_validation/ver-1e.md:105]: Just update it!

File: test/tests/ver-1e/comparison_ver-1e.py [hunk 12]
```diff
@@ -125,14 +130,57 @@ def get_lambdas(k,l,a):
 ax.set_xlabel(u'Time (s)')
 ax.set_ylabel(r"Concentration (moles/m$^3$)")
 ax.legend(loc="best")
-ax.set_xlim(0,1.0)
-ax.set_ylim(-1,20)
+ax.set_xlim(0,0.3)
```
Comment (simopier) [inline test/tests/ver-1e/comparison_ver-1e.py:133]:
```suggestion
ax.set_xlim(0,0.4)
```

File: test/tests/ver-1e/comparison_ver-1e.py [hunk 13]
```diff
@@ -125,14 +130,57 @@ def get_lambdas(k,l,a):
...
+ax.set_ylim(-0.2,1)
```
Comment (simopier) [inline test/tests/ver-1e/comparison_ver-1e.py:134]:
```suggestion
ax.set_ylim(-0.2,1.5)
```

File: test/tests/ver-1e/comparison_ver-1e.py [hunk 14]
```diff
@@ -26,74 +35,70 @@
...
-        f = 1/tan(lambda_range*a)
-        g = 1/tan(k*l*lambda_range)/k
-        idx = np.where(np.diff(np.sign(f+g)))
-        lambdas = np.expand_dims(lambda_range[idx][::2],axis=0)
-        return lambdas
-
-t0 = 0.2
-# TMAP 4 Analytical solution
+# Analytical parameters
+t0 = 0.01
```
Comment (simopier) [inline test/tests/ver-1e/comparison_ver-1e.py:47]:
```suggestion
t0 = 0.2
```

File: doc/content/verification_and_validation/ver-1e.md [hunk 15]
```diff
@@ -30,54 +34,79 @@ where
...
        image_name=ver-1e_comparison_time_closeup.png
        style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
        id=ver-1e_comparison_time_zoomed
-       caption=Zoomed-in view of comparison of TMAP8 calculation with the analytical solution for $t$ < 1 s. The analytical solution shows unphysical predictions close to $t$ = 0 s, whereas TMAP8's predictions are reasonable.
+       caption=Zoomed-in view of comparison of TMAP8 calculation with the analytical solution for $t$ < 0.3 s. The analytical solution shows unphysical predictions close to $t$ = 0 s, whereas TMAP8's predictions are reasonable.
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1e.md:109]:
```suggestion
       caption=Zoomed-in view of comparison of TMAP8 calculation with the analytical solution for $t$ < 0.4 s. The analytical solution shows unphysical predictions close to $t$ = 0 s, whereas TMAP8's predictions are reasonable.
```

File: doc/content/verification_and_validation/ver-1e.md [hunk 16]
```diff
@@ -30,54 +34,79 @@ where
...
 
-The error is calculated between the TMAP8 and analytical solution values after $t$ = 0.2 s. This is in order to ignore the unphysical predictions of the analytical solution at very small times as shown in [ver-1e_comparison_time_zoomed], which is a close-up view of [ver-1e_comparison_time] close to the start of the simulation.
+The error is calculated between the TMAP8 and analytical solution values after $t$ = 0.01 s. This is in order to ignore the unphysical predictions of the analytical solution at very small times as shown in [ver-1e_comparison_time_zoomed], which is a close-up view of [ver-1e_comparison_time] close to the start of the simulation. The departure from physical results at low $t$ is due to the limited number of $\lambda_n$ values being used from [eqn:roots]. We use the $\lambda_n$ values from 0 to 100 in this case. The $\lambda_n$ values in larger range will only have limited improvement in the accuracy but increase the running time for the computation of the analytical solutions.
```
Comment (simopier) [inline doc/content/verification_and_validation/ver-1e.md:103]:
```suggestion
The error is calculated between the TMAP8 and analytical solution values after $t$ = 0.2 s. This is in order to ignore the unphysical predictions of the analytical solution at very small times as shown in [ver-1e_comparison_time_zoomed], which is a close-up view of [ver-1e_comparison_time] close to the start of the simulation. The departure from physical results at low $t$ is due to the limited number of $\lambda_n$ values being used from [eqn:roots]. We use the $\lambda_n$ values from 0 to 100 in this case. The $\lambda_n$ values in larger range will only have limited improvement in the accuracy but increase the running time for the computation of the analytical solutions.
```

**Review by simopier — CHANGES_REQUESTED** [review]
I'm sorry to keep asking for changes, but given the poor results provided by the analytical solution in the zoomed-in graph, I think we should use a higher time threshold to start the comparison. 

Overall, this is a great improvement to the case!

**Comment by lin-yang-ly** [comment]
> I'm sorry to keep asking for changes, but given the poor results provided by the analytical solution in the zoomed-in graph, I think we should use a higher time threshold to start the comparison.
> 
> Overall, this is a great improvement to the case!

Okay, the time threshold of 0.2 make sense to me!

**Review by simopier — APPROVED** [review]
Looks great, thank you @lin-yang-ly!

**Pull request merged**